File: services/lucy-dotnet/src/Lucy.AuthPayment.Api/Modules/Payment/Models/WalletTransaction.cs
namespace lucy_auth_payment.Modules.Payment.Models;

public class WalletTransaction
{
    public string WalletTransactionId { get; set; } = null!;
    public string WalletId { get; set; } = null!;
    public string UserId { get; set; } = null!;
    public string? RelatedUserId { get; set; }
    public string TransactionType { get; set; } = null!;
    public string Direction { get; set; } = null!;
    public decimal Amount { get; set; }
    public decimal BalanceBefore { get; set; }
    public decimal BalanceAfter { get; set; }
    public string? RelatedRefType { get; set; }
    public string? RelatedRefId { get; set; }
    public string? DescriptionText { get; set; }
    public string TransactionStatus { get; set; } = "SUCCESS";
    public DateTime CreatedAt { get; set; } = DateTime.Now;
}

File: lucy-web-stack/lucy-backend/src/main/java/com/lucy/backend/payment/entity/WalletTransaction.java
package com.lucy.backend.payment.entity;

import jakarta.persistence.*;
import lombok.Data;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "WalletTransactions")
@Data
public class WalletTransaction {
    @Id
    @Column(name = "WalletTransactionId", length = 50)
    private String walletTransactionId;

    @Column(name = "WalletId", length = 50, nullable = false)
    private String walletId;

    @Column(name = "UserId", length = 50, nullable = false)
    private String userId;

    @Column(name = "RelatedUserId", length = 50)
    private String relatedUserId;

    @Column(name = "TransactionType", length = 50, nullable = false)
    private String transactionType;

    @Column(name = "Direction", length = 10, nullable = false)
    private String direction;

    @Column(name = "Amount", precision = 12, scale = 2, nullable = false)
    private BigDecimal amount;

    @Column(name = "BalanceBefore", precision = 12, scale = 2, nullable = false)
    private BigDecimal balanceBefore = BigDecimal.ZERO;

    @Column(name = "BalanceAfter", precision = 12, scale = 2, nullable = false)
    private BigDecimal balanceAfter = BigDecimal.ZERO;

    @Column(name = "RelatedRefType", length = 50)
    private String relatedRefType;

    @Column(name = "RelatedRefId", length = 50)
    private String relatedRefId;

    @Column(name = "DescriptionText", length = 255)
    private String descriptionText;

    @Column(name = "TransactionStatus", length = 30, nullable = false)
    private String transactionStatus = "SUCCESS";

    @Column(name = "CreatedAt", updatable = false)
    private LocalDateTime createdAt = LocalDateTime.now();
}
